pkgs/applications/editors/notepad-next/default.nix

```diff
@@ -0,0 +1,38 @@
+{ mkDerivation, lib, fetchFromGitHub }:
+
+mkDerivation rec {
+  pname = "notepad-next";
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "dail8859";
+    repo = "NotepadNext";
+    rev = "v${version}";
+    sha256 = "sha256-8S00rqF9u2wFmKARlJwiJTHc2klmmUf4h0SXH8cBw04=";
+    fetchSubmodules = true;
+  };
+
+  # References
+  #  https://github.com/dail8859/NotepadNext/blob/master/doc/Building.md
+  #  https://github.com/dail8859/NotepadNext/issues/99
+  buildPhase = ''
+    mkdir -p $out/bin $out/share/applications
+    substituteInPlace ./src/NotepadNext/NotepadNext.pro --replace /usr $out
+    mkdir build
+    cd build
+    qmake ../src/NotepadNext.pro
```

Comment:
This can be done via
``` nix
{
  nativeBuildInputs = [ qmake ];
  qmakeTargets = [ "src/NotepadNext.pro" ];
}
```

If you also move the `substituteInPlace` into `postPatch`, you don't even need a custom `buildPhase` :)